### PR TITLE
Added support for uint16 and improved `replace_all` to handle bodies with length > 1

### PR DESCRIPTION
**Pull request by skeqiqevian**
This commit adds a precision for `uint16_t` to Exo, which was necessary to compare against the Halide blur benchmark. I also added the corresponding AVX2 instructions that were necessary for the blur effort. One important system change is that `replace_all` is now capable of handling instructions whose proc semantics are longer than a single statement.

### CODE REVIEW

File: src/exo/stdlib/scheduling.py [hunk 1]
```diff
@@ -269,19 +264,26 @@ def _replace_helper(proc, subprocs, mem_aware, once):
         i = 0
         while True:
             try:
+                block = proc.find(f"{pattern} #{i}").expand(0, len(body) - 1)
+                assert len(block) == len(
+                    body
+                ), "block is too close to the end of the body"
```
Comment (yamaguchi1024) [inline src/exo/stdlib/scheduling.py:270]: @skeqiqevian 
Is this supposed to be an error when the length of a block from the pattern is less than the length of the subproc body? If so, shouldn't this be SchedulingError? Try/catching the `AssertionError` in the code below may hide various internal compiler errors

Comment (skeqiqevian) [reply src/exo/stdlib/scheduling.py:270]: As it currently works, `expand` will automatically clamp to the end of the scope. So if `proc.find` returns a statement at the end of a body, and we try to expand it by 2, we will receive a block of size 1. Maybe this should be a type of UnificationError?

Comment (yamaguchi1024) [reply src/exo/stdlib/scheduling.py:270]: Yeah I think `UnificationError` is reasonable here. If you change that, can you remove catching `AssertionError` from below?

**Review by yamaguchi1024 — APPROVED** [review]

File: src/exo/stdlib/scheduling.py [hunk 2]
```diff
@@ -269,19 +264,26 @@ def _replace_helper(proc, subprocs, mem_aware, once):
...
             except (TypeError, SchedulingError) as e:
                 if "failed to find matches" in str(e):
                     break
                 raise
-            except (_UnificationError, MemoryError, NotImplementedError):
+            except (
+                _UnificationError,
+                MemoryError,
+                NotImplementedError,
+                AssertionError,
```
Comment (yamaguchi1024) [inline src/exo/stdlib/scheduling.py:285]: Oh wait you wanna remove this

**Review by yamaguchi1024 — CHANGES_REQUESTED** [review]
Don't catch AssertionError

**Review by yamaguchi1024 — APPROVED** [review]

**Pull request merged**